Targets/API/Sources/HTTPMethod.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  TiclemoaRequest.swift
+//  API
+//
+//  Created by 김용우 on 2022/12/16.
+//  Copyright © 2022 nyongnyong. All rights reserved.
+//
+
+import Foundation
+
+enum HTTPMethod: String {
+    case get = "GET"
+    case post = "POST"
+    case put = "PUT"
+    case delete = "DELETE"
+    case patch = "PATCH"
+}
+
+struct TiclemoaURL {
+    let url: URL
+    let path: String
+    let queries: [String: String]
+    
+    func makeURL() -> URL {
+        return {
+            var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false)
+            urlComponents?.queryItems = queries.map({ URLQueryItem(name: $0, value: $1) })
+            return urlComponents?.url ?? URL(fileURLWithPath: "FailureURL")
+        }()
+    }
```

Comment:
```suggestion
    func makeURL() -> URL? {
        var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false)
        urlComponents?.queryItems = queries.map({ URLQueryItem(name: $0, value: $1) })
        return urlComponents?.url
    }
```

`FailureURL` 보다는 `nil, throw Error` 등을 활용하는게 어떨까요?

Reply:
수정했습니다